File: dumpinfo.pl
#!/usr/bin/perl -w

#---------------------------------------------------------------------
# dumpinfo (C) 2011 Gokhan Atil http://www.gokhanatil.com
#---------------------------------------------------------------------

use strict;

print "\ndumpinfo (C) 2011 Gokhan Atil http://www.gokhanatil.com \n\n";

($#ARGV == 0) or die "Usage: dumpinfo.pl filename\n\n";

open FILE, $ARGV[0] or die "Couldn't open file: $!\n\n";
binmode FILE;

my $buffer = '';
read( FILE, $buffer, 600 );

my ($magic1) = unpack 'x477C', $buffer;
my ($magic2) = unpack 'x3C', $buffer;

if ($magic1 == 49 ) {

my ($filevermajor,$filevermin) = unpack 'CC', $buffer;
my ($year,$mon,$day,$hour,$min,$sec) = unpack 'x41nCCCCC', $buffer;
my ($version) = unpack 'x476a14', $buffer;
my ($platform) = unpack 'x132A30', $buffer;
my ($charset) = unpack 'x294A20', $buffer;
my ($blocksize) = unpack 'x37n', $buffer;
my ($jobname) = unpack 'x66A40', $buffer;
my ($filevernum) = unpack 'n', $buffer;
my ($charsetID) = unpack 'x40C', $buffer;
my ($mastertablepos) = unpack 'x57C', $buffer;
my ($mastertablelen) = unpack 'x62N', $buffer;
my ($jguid) = unpack 'x15H32', $buffer;

print " ........Filetype = Datapump dumpfile\n"; 
print " ......DB Version = $version \n";
print " File Version Str = $filevermajor.$filevermin \n";
print " File Version Num = $filevernum \n";
print " ........Job Guid = $jguid \n";
print " Master Table Pos = $mastertablepos \n";
print " Master Table Len = $mastertablelen \n";
print " ......Charset ID = $charsetID \n";
print " ...Creation date = $mon-$day-$year $hour:$min:$sec \n";
print " ........Job Name = $jobname \n";
print " ........Platform = $platform \n";
print " ........Language = $charset \n";
print " .......Blocksize = $blocksize \n";

} elsif ($magic2 == 69  ) {

my ($charset ) = unpack 'x2C', $buffer;
my ($exportdate) = unpack 'x106a23', $buffer;
my ($exportver) = unpack 'x11a8', $buffer;

print " ........Filetype = Classic Export file\n"; 
print " ..Export Version = $exportver \n";
print " .....Direct Path = 0 (Conventional Path) \n";
print " .Characterset ID = $charset \n";
print " ...Creation date = $exportdate \n";


} elsif ($magic2 == 68 ) {

my ($charset ) = unpack 'x2C', $buffer;
my ($exportdate) = unpack 'x105a23', $buffer;
my ($exportver) = unpack 'x13a8', $buffer;

print " ........Filetype = Classic Export file\n"; 
print " ..Export Version = $exportver \n";
print " .....Direct Path = 1 (Direct Path) \n";
print " .Characterset ID = $charset \n";
print " ...Creation date = $exportdate \n";


} else {

print " ..........Error = Unsupported File \n";

}

print "\n";

close FILE

File: dumpinfo.py
#!/usr/bin/python

#---------------------------------------------------------------------
# dumpinfo (C) 2014 Gokhan Atil http://www.gokhanatil.com
#---------------------------------------------------------------------

import struct, binascii, sys

if len(sys.argv) != 2:
    print 'Usage: dumpinfo.py filename'
    exit()


f = open(sys.argv[1], 'rb')

try:
    buffer = f.read(600)

finally:
    f.close()    

magic1 = struct.unpack_from( 'B', buffer, 477 )[0]
magic2 = struct.unpack_from( 'B', buffer, 3 )[0]

print magic1
print magic2

if (magic1==49):
    filevermajor,filevermin = struct.unpack_from( 'BB', buffer )
    year,mon,day,hour,min,sec = struct.unpack_from( 'HBBBBB', buffer, 41 )
    version = struct.unpack_from( '14s', buffer, 477 )[0]
    platform = struct.unpack_from( '30s', buffer, 133 )[0]
    charset = struct.unpack_from( '20s', buffer, 295 )[0]
    blocksize1,blocksize2 = struct.unpack_from( 'BB', buffer, 37 )
    jobname = struct.unpack_from( '40s', buffer, 67 )[0]
    filevernum = struct.unpack_from( 'H', buffer )[0]
    charsetID = struct.unpack_from( 'B', buffer, 40 )[0]
    mastertablepos = struct.unpack_from( 'B', buffer, 57 )[0]
    len1,len2,len3,len4 = struct.unpack_from( 'BBBB', buffer, 62 )
    jguid = struct.unpack_from( '16s', buffer, 15 )[0]

    # fix network byte order
    
    blocksize = blocksize1 * 256 + blocksize2   
    mastertablelen = (len1 * 256 + len2) * 65536 + (len3 * 256 + len4)

    print ' ........Filetype = Datapump dumpfile'
    print ' ......DB Version = ' + version
    print ' File Version Str = ' + str(filevermajor) + '.' + str(filevermin) 
    print ' File Version Num = ' + str(filevernum) 
    print ' ........Job Guid = ' + binascii.hexlify(jguid)
    print ' Master Table Pos = ' + str(mastertablepos) 
    print ' Master Table Len = ' + str(mastertablelen)
    print ' ......Charset ID = ' + str(charsetID)
    print ' ...Creation date = ' + str(day) + '-' + str(mon) + '-' + str(year) + ' ' + str(hour) + ':'+ str(min) + ':'+ str(sec) 
    print ' ........Job Name = ' + jobname
    print ' ........Platform = ' + platform
    print ' ........Language = ' + charset
    print ' .......Blocksize = ' + str(blocksize)
    
elif (magic2 == 69):
    exportdate = struct.unpack_from( '20s', buffer, 108 )[0]
    exportver = struct.unpack_from( '8s', buffer, 11 )[0]
    print ' ........Filetype = Classic Export file'
    print ' ..Export Version = ' + str(exportver)
    print ' .....Direct Path = 0 (Conventional Path)'
    print ' ...Creation date = ' + exportdate

elif (magic2 == 68):
    exportdate = struct.unpack_from( '20s', buffer, 109 )[0]
    exportver = struct.unpack_from( '8s', buffer, 13 )[0]
    print ' ........Filetype = Classic Export file'
    print ' ..Export Version = ' + str(exportver)
    print ' .....Direct Path = 1 (Direct Path)'
    print ' ...Creation date = ' + exportdate

else:
    print ' ..........Error = Unsupported File';

